lib/ex_format/lines.ex

```diff
@@ -0,0 +1,15 @@
+defmodule ExFormat.Lines do
+  def initialize_lines_store(string) do
+  	lines = String.split(string, "\n")
+    Agent.start_link(fn -> %{} end, name: :lines)
+    for {line, i} <- Enum.with_index(lines) do
+      update_line(i + 1, String.trim(line))
+    end
```

Comment:
You can't "override" an agent. Once you start an agent called `:lines`, starting a new one called `:lines` will result in an error, that is, `start_link` will return `{:error, _}` instead of `{:ok, _}`. But you never look at the return value of `start_link`, so this goes unnoticed. For now, the solution is to stop the agent after formatting with something like


```elixir
def format(string) do
  ...
after
  Agent.stop(:lines)
end
```

but this will still fail with concurrent calls to `format`. The real problem is that we are solving an extremely functional and pure problem, but you are using a substantially destructive and mutable tool (agent). The solution is to drop agents completely and put lines in the formatter's state, alongside everything else. Same goes for comments.